### PR TITLE
Allow using multiple host disks in the same directory

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

If two disks are in the same host directory, don't mount the twho
resulting host directories in the same place in the container. Instead
every mount point for every disk is now unique, based on the volume
name.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1726766

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mfranczy** [comment]
I run functional tests locally and sometimes some of them fail, just an example:
```
• Failure [38.661 seconds]
Storage
/root/go/src/kubevirt.io/kubevirt/tests/storage_test.go:54
  Starting a VirtualMachineInstance
  /root/go/src/kubevirt.io/kubevirt/tests/storage_test.go:64
    [rfe_id:2298][crit:medium][vendor:cnv-qe@redhat.com][level:component] With HostDisk and PVC initialization
    /root/go/src/kubevirt.io/kubevirt/tests/storage_test.go:301
      With smaller than requested PVCs
      /root/go/src/kubevirt.io/kubevirt/tests/storage_test.go:474
        Should initialize an empty PVC with a disk.img when disk is too small but within toleration [It]
        /root/go/src/kubevirt.io/kubevirt/tests/storage_test.go:546

        Timed out after 30.000s.
        Timed out waiting for VMI to enter [Running] phase(s)
        Expected
            <v1.VirtualMachineInstancePhase>: Scheduled
        to be an element of
            <[]interface {} | len:1, cap:1>: [["Running"]]

        /root/go/src/kubevirt.io/kubevirt/tests/utils.go:2121
```
I am checking if this is related to change or my something is wrong with my cluster...

**Comment by rmohr** [comment]
Interesting. Can you have a look on the virt-handler logs? It looks like the pod was created and running properly, but that the VMI did not start. There you normally should see something in `kubectl get events` and the  virt-handler logs.

File: tests/storage_test.go [hunk 1]
```diff
@@ -321,9 +321,32 @@ var _ = Describe("Storage", func() {
 							virtClient,
 							vmiPod,
 							vmiPod.Spec.Containers[0].Name,
-							[]string{"find", hostDiskDir, "-name", diskName, "-size", "1G"},
+							[]string{"find", hostdisk.GetMountedHostDiskDir("host-disk"), "-name", diskName, "-size", "1G"},
 						)
-						Expect(strings.Contains(output, diskPath)).To(BeTrue())
+						Expect(output).To(ContainSubstring(hostdisk.GetMountedHostDiskPath("host-disk", diskPath)))
+
+						err = virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})
```
Comment (dhiller) [inline tests/storage_test.go]: This looks like cleanup, shouldn't this be done in `AfterEach`?

Comment (rmohr) [reply tests/storage_test.go]: let me check.

Comment (rmohr) [reply tests/storage_test.go]: Yep you are right. Moved the whole cleanup sequence into an `AfterEach`.

File: tests/storage_test.go [hunk 2]
```diff
@@ -321,29 +339,39 @@ var _ = Describe("Storage", func() {
...
+						vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+						nodeName = vmiPod.Spec.NodeName
+						output, err := tests.ExecuteCommandOnPod(
+							virtClient,
+							vmiPod,
+							vmiPod.Spec.Containers[0].Name,
+							[]string{"find", hostdisk.GetMountedHostDiskDir("anotherdisk"), "-name", "another.img", "-size", "1G"},
+						)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(output).To(ContainSubstring(hostdisk.GetMountedHostDiskPath("anotherdisk", filepath.Join(hostDiskDir, "another.img"))))
```
Comment (slintes) [inline tests/storage_test.go]: Shouldn't we do this check for both disks, just to make sure we didn't break the first one with this patch?

Other than that lgtm

Comment (rmohr) [reply tests/storage_test.go]: Done

**Comment by rmohr** [comment]
@slintes ok, the test should finally succeed.

**Comment by slintes** [comment]
only 1 unrelated test failure on all lanes :slightly_smiling_face: 

/lgtm
/approve

File: tests/storage_test.go [hunk 3]
```diff
@@ -321,29 +339,49 @@ var _ = Describe("Storage", func() {
...
 							vmiPod.Spec.Containers[0].Name,
-							[]string{"find", hostDiskDir, "-name", diskName, "-size", "1G"},
+							[]string{"find", hostdisk.GetMountedHostDiskDir("host-disk"), "-name", diskName, "-size", "1G"},
 						)
-						Expect(strings.Contains(output, diskPath)).To(BeTrue())
+						Expect(err).ToNot(HaveOccurred())
+						Expect(output).To(ContainSubstring(hostdisk.GetMountedHostDiskPath("host-disk", diskPath)))
+					})
+
+					It("should start with multiple hostdisks in the same directory", func() {
```
Comment (nellyc) [inline tests/storage_test.go]: @ILpinto do we have a test ID for this one?

**Pull request merged**